.orchestra/ci/ci-hook/app.py

```diff
@@ -89,77 +159,309 @@ def __exit__(self, exc_type, exc_val, exc_tb):
+                variables["BASE_USER_OPTIONS_YML"] = pusher_user_options
+                variables["SSH_PRIVATE_KEY"] = revng_push_ci_private_key
+                variables["PUSH_CHANGES"] = "1"
+            else:
+                tpl_params = {
+                    "clone_namespace": "/".join(repo_url.split("/")[:-1]),
+                    "private_sources": "",
+                    "private_bin_archives": ""
+                }
+                if not is_anonymous:
+                    # Placeholders are replaced in the shell script since we don't want to intentionally introduce shell
+                    # injection vulnerabilities.
+                    tpl_params.update({
+                        "private_sources": "- private: %PRIVATE_SOURCES_PLACEHOLDER%",
+                        "private_bin_archives": "- private: %PRIVATE_BIN_ARCHIVES_PLACEHOLDER%"
+                    })
+
+                variables["BASE_USER_OPTIONS_YML"] = \
+                    string.Template(pull_request_user_options).substitute(tpl_params)
+
+            parameters = {
+                "ref": BRANCH,
+                "variables": [{"key": key, "value": value}
+                              for key, value
+                              in variables.items()]
+            }
+            print(json.dumps(parameters, indent=2))
+            pipeline = project.pipelines.create(parameters)
+
+    except gitlab.exceptions.GitlabCreateError:
+        # If the user didn't have the permissions to create a new pipeline, use the default unprivileged user instead
+        if username != default_user:
+            traceback.print_exc()
+            print("Running as default user")
+            return trigger_ci(default_user, repo_url, base_repo_url, ref, status_update_metadata)
+        raise
+
+
+LAB_TO_HUB_STATUS_MAP = {
+    "created": "queued",
+    "waiting_for_resource": "queued",
+    "preparing": "queued",
+    "pending": "queued",
+    "running": "in_progress",
+    "success": "completed",
+    "failed": "completed",
+    "canceled": "completed",
+    "skipped": "completed",
+    "manual": "queued",
+    "scheduled": "queued"
+}
+
+LAB_TO_LAB_STATUS_MAP = {
+    "created": "started",
+    "waiting_for_resource": None,
+    "preparing": None,
+    "pending": None,
+    "running": None,
+    "success": "completed",
+    "failed": "failed",
+    "canceled": "cancelled",
+    "skipped": "skipped",
+    "manual": None,
+    "scheduled": None
+}
+
+LAB_TO_HUB_CONCLUSION_MAP = {
+    "success": "success",
+    "failed": "failure",
+    "skipped": "skipped",
+    "canceled": "cancelled",
+}
+
+
+def get_pl_metadata(admin_gl, pipeline_id):
+    pipeline = admin_gl.projects.get(PROJECT_ID).pipelines.get(pipeline_id)
+    variables = pipeline.variables.list()
+
+    for var in variables:
+        if var.key == "REVNG_CI_STATUS_UPDATE_METADATA":
+            return json.loads(var.value)
+
+
+@app.route('/ci-hook/gitlab_ci', methods=["POST"])
```

Comment:
```suggestion
@app.route('/ci-hook/gitlab-ci', methods=["POST"])
```

Can you rename the method and URL to be more explicit on the fact that this handles the result of a CI job/pipeline?
I was initially confused and thought we no longer triggered a pipeline upon a push on gitlab.